operator/e2e_testing/utils/k8s_client.go

```diff
@@ -0,0 +1,335 @@
+	GVR       schema.GroupVersionResource
+}
+
+// ApplyYAMLFile applies a YAML file containing Kubernetes resources
+// namespace parameter is optional - pass empty string to use namespace from YAML
+func ApplyYAMLFile(ctx context.Context, yamlFilePath string, namespace string, restConfig *rest.Config, logger *logrus.Logger) ([]AppliedResource, error) {
+	logger.Debugf("📄 Applying resources from %s...\n", yamlFilePath)
+
+	// Read the YAML file
+	yamlData, err := os.ReadFile(yamlFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read YAML file %s: %w", yamlFilePath, err)
+	}
+
+	return applyYAMLData(ctx, yamlData, namespace, restConfig, logger)
+}
+
+// WaitForPods waits for pods to be ready in the specified namespaces
+// labelSelector is optional (pass empty string for all pods), timeout of 0 defaults to 5 minutes, interval of 0 defaults to 5 seconds
+func WaitForPods(ctx context.Context, restConfig *rest.Config, namespaces []string, labelSelector string, timeout time.Duration, interval time.Duration, logger *logrus.Logger) error {
+	clientset, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return fmt.Errorf("failed to create clientset: %w", err)
+	}
+
+	timeoutCtx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+
+	// If no namespaces specified, use default
+	if len(namespaces) == 0 {
+		namespaces = []string{"default"}
+	}
+
+	logger.Debugf("⏳ Waiting for pods to be ready in namespaces: %v", namespaces)
+
+	return wait.PollUntilContextTimeout(timeoutCtx, interval, timeout, true, func(ctx context.Context) (bool, error) {
+		allReady := true
+		totalPods := 0
+		readyPods := 0
+
+		for _, namespace := range namespaces {
+			pods, err := clientset.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{
+				LabelSelector: labelSelector,
+			})
+			if err != nil {
+				logger.Errorf("Failed to list pods in namespace %s: %v", namespace, err)
+				return false, nil
+			}
+
+			for _, pod := range pods.Items {
+				totalPods++
+				if isPodReady(&pod) {
+					readyPods++
+				} else {
+					allReady = false
+				}
+			}
+		}
+
+		if totalPods == 0 {
+			logger.Debug("⏳ No pods found yet, resources may still be creating pods...")
+			return false, nil
+		}
+
+		if !allReady {
+			logger.Debugf("⏳ Waiting for %d more pods to become ready...", totalPods-readyPods)
+		}
+
+		return allReady, nil
+	})
+}
+
+// ApplyYAMLContent applies YAML content directly to Kubernetes
+func ApplyYAMLContent(ctx context.Context, yamlContent string, namespace string, restConfig *rest.Config, logger *logrus.Logger) ([]AppliedResource, error) {
+	logger.Debug("📄 Applying YAML content...")
+	return applyYAMLData(ctx, []byte(yamlContent), namespace, restConfig, logger)
+}
+
+// applyYAMLData is the common function that applies YAML data to Kubernetes
+func applyYAMLData(ctx context.Context, yamlData []byte, namespace string, restConfig *rest.Config, logger *logrus.Logger) ([]AppliedResource, error) {
```

Comment:
`yamlData` can be a `string` directly, since we read the file into `[]byte`, and then all further usage of that data is in `string` form anyway. The caller `ApplyYAMLFile` can pass `string` data directly.

```suggestion
func applyYAMLData(ctx context.Context, yamlData, namespace string, restConfig *rest.Config, logger *logrus.Logger) ([]AppliedResource, error) {
```

Reply:
Done